tools/codegen/operator_versions/gen_mobile_upgraders.py

```diff
@@ -0,0 +1,357 @@
+            convert_constant = "true" if constant_from_yaml else 'false'
+        else:
+            raise ValueError(
+                f"The type of {constant_from_yaml} is {type(constant_from_yaml)}. "
+                "Please add change in construct_constants function in gen_mobile_upgraders.py.")
+        constants_list_part.append(
+            ONE_CONSTANT.substitute(
+                constant=convert_constant
+            )
+        )
+    return CONSTANT_LIST.substitute(constant_list="".join(constants_list_part))
+
+def construct_operators(operator_list_from_yaml: List) -> str:
+    operator_list_part = []
+    for operator in operator_list_from_yaml:
+        operator_list_part.append(
+            ONE_OPERATOTR_STRING.substitute(
+                operator_name=operator[0],
+                overload_name=operator[1],
+                num_of_args=operator[2],
+            )
+        )
+    return OPERATOR_STRING_LIST.substitute(operator_string_list="".join(operator_list_part))
+
+def construct_types(types_tr_list_from_yaml: List) -> str:
+    types_tr_list_part = []
+    for types_tr in types_tr_list_from_yaml:
+        types_tr_list_part.append(
+            ONE_TYPE.substitute(
+                type_str=types_tr
+            )
+        )
+    return TYPE_LIST.substitute(type_list="".join(types_tr_list_part))
+
+def construct_register_size(register_size_from_yaml: int) -> str:
+    if (not isinstance(register_size_from_yaml, int)):
+        raise ValueError(
+            f"Input register size is {register_size_from_yaml} and"
+            "it's type is {type(register_size_from_yaml)}. An int type is expected.")
+    return str(register_size_from_yaml)
+
+def construct_one_operator_in_version_map(operator_name: str, upgrader_list: List) -> str:
+    upgraders_in_version_map_part = []
+    for one_upgrader in upgrader_list:
+        upgraders_in_version_map_part.append(
+            ONE_UPGRADER_IN_VERSION_MAP.substitute(
+                upgrader_min_version=one_upgrader[0],
+                upgrader_max_version=one_upgrader[1],
+                upgrader_name=one_upgrader[2],
+                bytecode_func_index=one_upgrader[3]
+            )
+        )
+    return ONE_OPERATOR_IN_VERSION_MAP.substitute(
+        operator_name=operator_name,
+        upgrader_list_in_version_map="".join(upgraders_in_version_map_part)
+    )
+
+def construct_version_maps(upgrader_bytecode_function_to_index_map: Dict) -> Dict:
+    version_map = torch._C._get_operator_version_map()
+    operator_list_in_version_map_part = []
+    for op_name in version_map:
+        upgraders_in_version_map_part = []
+        for upgrader_entry in version_map[op_name]:
+            # Split a string by "_" and filter empty string in the list
+            # For example: "div__Scalar_0_3" => ['div', 'Scalar', '0', '3']
+            upgrader_info = list(filter(lambda token: token != "", upgrader_entry.upgrader_name.split('_')))
```

Comment:
I think it is not safe to rely on the upgrader name to figure out the start and end. I think it rather should be based on version bump value. So for example, if the version bump happened at 4, we know that the range will be from 0 to 3.